### ISSUE TITLE
worker is stuck idle when building is a valid activity

### ISSUE THREAD

**Issue by karlthepagan** [ScreepsOCS/screeps.behaviour-action-pattern#1311]
Starting here with the invasion, but even after the room situation changes my worker is still dead idle.
https://screeps.com/a/#!/history/E1N87?t=18731594

[9:23:56 AM]18732043 Creep run {"creepName":"worker-2000-4","pos":{"x":19,"y":34,"roomName":"E1N87"},"Behaviour":"worker","Creep":"run"}

**Comment by karlthepagan** [comment]
As soon as a new creep spawned he began working.

This might not be a bug, but he was not flashing the whistle/idle say.

**Comment by karlthepagan** [comment]
I switched the say to public and other players are able to see what the creeps are saying.

... it was my display options.

**Comment by karlthepagan** [comment]
Going to reopen this because I was able to catch a worker in this room idle again.

**Comment by chobobobo** [comment]
I don't see in the replay where there is a construction site for that creep to build?

**Comment by karlthepagan** [comment]
There's some labs above it near the middle. I don't think I have an easy fix. This might be resolved with your denounce change in 0.7.1

**Comment by karlthepagan** [comment]
Sorry in the first replay it was dismantling which is an inflow action not an outflow like building. That might mean this is 2 bugs.

**Comment by chobobobo** [comment]
Yeah, I'll see if I can get the room refactor finished so I can put in the behaviour refactor, once that's done the worker debounce will be easy.

**Comment by karlthepagan** [comment]
I think this is only a temporary issue and it has been very difficult to repro.
